### PR TITLE
Support auto-import in macro

### PR DESCRIPTION
**Pull request by edwin0cheng**
Fixed: #3854

### CODE REVIEW

File: crates/ra_assists/src/handlers/auto_import.rs [hunk 1]
```diff
@@ -45,15 +45,12 @@ pub(crate) fn auto_import(ctx: AssistCtx) -> Option<Assist> {
...
     let mut group = ctx.add_assist_group(auto_import_assets.get_import_group_message());
     for import in proposed_imports {
         group.add_assist(AssistId("auto_import"), format!("Import `{}`", &import), |edit| {
-            edit.target(auto_import_assets.syntax_under_caret.text_range());
-            insert_use_statement(
-                &auto_import_assets.syntax_under_caret,
-                &import,
-                edit.text_edit_builder(),
-            );
+            edit.target(range);
```
Comment (edwin0cheng) [inline crates/ra_assists/src/handlers/auto_import.rs:52]: I can't get the `AssistContext` in this closure, so I pass the `ctx` to `ActionBuilder`. Another options is passing the `ctx` directly as closure arguments, but I don't want to change too much code. I will prefer to fix it in other PR.

**Comment by matklad** [comment]
bors r=SomeoneToIgnore

huh, I thought you have r+ rights? Re-added to bors anyway.

File: crates/ra_assists/src/handlers/auto_import.rs (hunk 1 above)
Comment (SomeoneToIgnore) [reply crates/ra_assists/src/handlers/auto_import.rs:52]: I'm currently trying to implement https://github.com/rust-analyzer/rust-analyzer/issues/1062 and looks like I will have to change it in order to use the `insert_use_statement` in `ra_ide`.

So give it a couple of weeks before doing the refactoring :)

**Pull request merged**